### PR TITLE
Reorganize src and test folder structure

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Move `src/Tests` to `tests` folder. It can separate the `tests` from `src` folder.
- Add the `autoload-dev` block in `composer.json` to define the test classes autoload.
- Add the white filter tag in `phpunit.xml` to define the code coverage lists.

### CODE REVIEW

**Comment by opulencephp** [comment]
Hey @peter279k, thanks for opening the PR!  The reason I do not have the tests in a root-level `tests` directory is because this repo will _probably_ become a read-only subtree split of the main Opulence repository.  In there, I keep tests under `src/{Library name}/Tests` directory so that someone could checkout a library's read-only repository and run the unit tests within it.

**Comment by peter279k** [comment]
Hi @opulencephp, thank you for your reply. I will move the `Tests` to `src` folder back,

**Comment by peter279k** [comment]
@opulencephp, I've moved the `tests` to `src/Tests` folder :+1: .
Please check that and review the PR.

File: composer.json [hunk 1]
```diff
@@ -1,44 +1,46 @@
...
+      "role": "Architect"
+    }
+  ],
+  "license": "MIT",
+  "autoload": {
+    "psr-4": {
+      "Opulence\\Net\\": "src"
+    }
+  },
+  "autoload-dev": {
```
Comment (opulencephp) [inline composer.json]: This probably isn't necessary anymore now that we moved the tests back under `src/Tests`

File: phpunit.xml [hunk 2]
```diff
@@ -4,4 +4,12 @@
     <testsuite name="Net">
         <directory>src/Tests</directory>
     </testsuite>
-</phpunit>
\ No newline at end of file
+     <filter>
```
Comment (opulencephp) [inline phpunit.xml]: I'm not familiar with this feature. What exactly is this doing?

Comment (peter279k) [reply phpunit.xml]: This can define the code coverage folder to tell `PHPUnit` where can do that.

File: composer.json (hunk 1 above)
Comment (peter279k) [reply composer.json]: I think this block should be existed because the namespace is different.

Comment (opulencephp) [reply composer.json]: The test namespace will be resolved by the `autoload` psr-4 config because the tests are in the `src/Tests` directory.

Comment (peter279k) [reply composer.json]: @opulencephp, thank you for your reply. You're right and I remove the `autoload-dev` block in latest commit now :-).

**Pull request merged**